frontend/src/scenes/dashboard/dashboardLogic.tsx

```diff
@@ -1491,7 +1540,12 @@ export const dashboardLogic = kea<dashboardLogicType>([
                     // this is done in the reducer for dashboard
                 } else if (source === DashboardEventSource.DashboardHeaderSaveDashboard) {
                     // save edit mode changes
-                    actions.setFiltersAndLayoutsAndVariables(values.temporaryFilters, values.temporaryVariables)
+                    actions.setFiltersAndLayoutsAndVariables(
```

Comment:
This action should probably be renamed to something like `saveEditModeChanges`. Keeping this as a follow-up task to avoid further enlarging this PR.